test/test_grid.py

```diff
@@ -116,6 +116,31 @@ def test_init_grid_var_attrs(self):
         self.assertEqual(n_face_nodes, grid.nMaxMesh2_face_nodes)
 
     # def test_init_dimension_attrs(self):
+    def test_build_face_edges_connectivity(self):
+        """Tests to see if the generated face_edges_connectivity number match
+        the calculated results from Euler formular."""
+        ug_filename_list = [
+            self.ug_filename1, self.ug_filename2, self.ug_filename3
+        ]
+        for ug_file_name in ug_filename_list:
+            ug_filename1 = current_path / "meshfiles" / ug_file_name
+            xr_tgrid1 = xr.open_dataset(ug_filename1)
+            tgrid1 = ux.Grid(xr_tgrid1)
+            mesh2_face_nodes = tgrid1.ds["Mesh2_face_nodes"]
+            tgrid1.build_face_edges_connectivity()
+            mesh2_face_edges = tgrid1.ds.Mesh2_face_edges
+            mesh2_edge_nodes = tgrid1.ds.Mesh2_edge_nodes
+
+            # Assert if the mesh2_face_edges sizes are correct.
+            self.assertEqual(mesh2_face_edges.sizes["nMesh2_face"],
+                             mesh2_face_nodes.sizes["nMesh2_face"])
+            self.assertEqual(mesh2_face_edges.sizes["nMaxMesh2_face_edges"],
+                             mesh2_face_nodes.sizes["nMaxMesh2_face_nodes"])
+            self.assertEqual(mesh2_face_edges.sizes["Two"], 2)
```

Comment:
When I tried writing them as
`            # Assert if the mesh2_face_edges sizes are correct.
>           self.assertEqual(mesh2_face_edges.nMesh2_face,
                             mesh2_face_nodes.nMesh2_face)
`

It throws the following errors:
`
self = <xarray.DataArray 'nMesh2_face' (nMesh2_face: 5400)>
array([ True,  True,  True, ...,  True,  True,  True])
Dimensions without coordinates: nMesh2_face
`

Here I just need to compare the size of these two arrays, I wonder how should I solve these errors.

I also tried `mesh2_face_edges.nMesh2_face.sizes()`, and it throws `'Frozen' object is not callable`

Reply:
The attributes I mentioned above are through the `Grid` object, not through the face edges or node variables. They are set according to the UGRID conventions.

Reply:
> ```python
> nMaxMesh2_face_edges
> ```

I am still a bit confused here. So when I tried using
`
            self.assertEqual(tgrid1.nMaxMesh2_face_edges,
                             tgrid1.nMaxMesh2_face_nodes)
`
it will throw the following exception
`
            # Assert if the mesh2_face_edges sizes are correct.
>           self.assertEqual(tgrid1.nMaxMesh2_face_edges,
                             tgrid1.nMaxMesh2_face_nodes)
E           AttributeError: 'Grid' object has no attribute 'nMaxMesh2_face_edges'
`
Even though the `tgrid1.build_face_edges_connectivity()` is successfully called

Reply:
Let me take a look into it, that dimension might have not been explicitly set as an attribute (only ones from face nodes are iirc)

Reply:
> Let me take a look into it, that dimension might have not been explicitly set as an attribute (only ones from face nodes are iirc)

I tried adding the dimension at `__init_ds_var_names__` but it still doesn't work. I have push my changes in `grid.py` and pushed it. Thank you very much!

Reply:
Let's leave what you currently have until we have the UXarray redesign is merged. Currently `__init_grid_var_attrs__()` supports the following attributes. This explains why `nMaxMesh2_face_edges` doesn't work, but the following do. I'd like to extend this to include other attributes, such as the ones you are using. 
```Python
self.ds_var_names = {
            "Mesh2": "Mesh2",
            "Mesh2_node_x": "Mesh2_node_x",
            "Mesh2_node_y": "Mesh2_node_y",
            "Mesh2_node_z": "Mesh2_node_z",
            "Mesh2_face_nodes": "Mesh2_face_nodes",
            # initialize dims
            "nMesh2_node": "nMesh2_node",
            "nMesh2_face": "nMesh2_face",
            "nMaxMesh2_face_nodes": "nMaxMesh2_face_nodes"
        }

Reply:
> Let's leave what you currently have until we have the UXarray redesign is merged. Currently `__init_grid_var_attrs__()` supports the following attributes. This explains why `nMaxMesh2_face_edges` doesn't work, but the following do.
> 
> ```python
> self.ds_var_names = {
>             "Mesh2": "Mesh2",
>             "Mesh2_node_x": "Mesh2_node_x",
>             "Mesh2_node_y": "Mesh2_node_y",
>             "Mesh2_node_z": "Mesh2_node_z",
>             "Mesh2_face_nodes": "Mesh2_face_nodes",
>             # initialize dims
>             "nMesh2_node": "nMesh2_node",
>             "nMesh2_face": "nMesh2_face",
>             "nMaxMesh2_face_nodes": "nMaxMesh2_face_nodes"
>         }
> ```

Thank you so much for your help!
Just curious, I Also try adding the ` "nMaxMesh2_face_edges"` inside the `__init_grid_var_attrs__()` but it doesn't solve the problem. I wonder if you know why? Thanks!
> 
> ```python
> self.ds_var_names = {
>             "Mesh2": "Mesh2",
>             "Mesh2_node_x": "Mesh2_node_x",
>             "Mesh2_node_y": "Mesh2_node_y",
>             "Mesh2_node_z": "Mesh2_node_z",
>             "Mesh2_face_nodes": "Mesh2_face_nodes",
>             # initialize dims
>             "nMesh2_node": "nMesh2_node",
>             "nMesh2_face": "nMesh2_face",
>             "nMaxMesh2_face_nodes": "nMaxMesh2_face_nodes"
>            "nMaxMesh2_face_edges": "nMaxMesh2_face_edges",
>            "Two": "Two"
>         }
> ```